router-tests/jwks/jwks.go

```diff
@@ -0,0 +1,91 @@
+package jwks
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const (
+	jwtKeyID          = "123456789"
+	signingMethodType = "RSA" // This should match signingMethod below
+
+	jwksHTTPPath = "/.well-known/jwks.json"
+)
+
+var (
+	signingMethod = jwt.SigningMethodRS256
+)
+
+type Server struct {
+	privateKey *rsa.PrivateKey
+	httpServer *httptest.Server
+}
+
+func (s *Server) Close() {
+	s.httpServer.Close()
+}
+
+func (s *Server) Token(claims map[string]any) (string, error) {
+	token := jwt.NewWithClaims(signingMethod, jwt.MapClaims(claims))
+	token.Header["kid"] = jwtKeyID
+	return token.SignedString(s.privateKey)
+}
+
+type jsonWebKeySet struct {
+	Keys []jsonWebKey `json:"keys"`
+}
+
+type jsonWebKey struct {
+	Algorithm string `json:"alg"`
+	Curve     string `json:"crv"`
+	Exponent  string `json:"e"`
+	K         string `json:"k"`
+	ID        string `json:"kid"`
+	Modulus   string `json:"n"`
+	Type      string `json:"kty"`
+	Use       string `json:"use"`
+	X         string `json:"x"`
+	Y         string `json:"y"`
+}
+
+func (s *Server) jwksJSON(w http.ResponseWriter, r *http.Request) {
+	k := jsonWebKey{
+		Type:      signingMethodType,
+		Algorithm: signingMethod.Name,
+		Use:       "sig",
+		ID:        jwtKeyID,
+		Exponent:  base64.URLEncoding.EncodeToString(big.NewInt(int64(s.privateKey.E)).Bytes()),
+		Modulus:   base64.URLEncoding.EncodeToString(s.privateKey.N.Bytes()),
+	}
+	data, err := json.Marshal(jsonWebKeySet{Keys: []jsonWebKey{k}})
+	if err != nil {
+		panic(err)
```

Comment:
Just checking we definitely want to panic here.

Reply:
We do, this is just for testing